components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/LabelDAOImpl.java

```diff
@@ -367,6 +394,46 @@ private void deleteLabelAccessUrlMappings(String labelId) throws APIMgtDAOExcept
 
     }
 
+    /**
+     * Add default labels.
+     * 
+     * @throws APIMgtDAOException If an error occurred while adding labels.
+     */
+    static void initDefaultLabels() throws APIMgtDAOException {
+        if (!isLabelsExists()) {
+            //Todo : default labels need to be configurable
+            List<String> accessUrls = new ArrayList<>();
+            Label.Builder labelBuilder = new Label.Builder();
+            labelBuilder.id(UUID.randomUUID().toString());
+            labelBuilder.name(APIMgtConstants.DEFAULT_LABEL_NAME);
+            accessUrls.add(APIMgtConstants.DEFAULT_LABEL_ACCESS_URL);
+            labelBuilder.accessUrls(accessUrls);
+            addLabel(labelBuilder.build());
+        }
+    }
+
+    /**
+     * Checks if any labels added to DB and already available
+     *
+     * @return true if there are any labels available in the system
+     * @throws APIMgtDAOException If an error occurs while checking labels existence
+     */
+    private static boolean isLabelsExists() throws APIMgtDAOException {
+        final String query = "SELECT 1 FROM AM_LABELS";
+        try {
+            try (Connection connection = DAOUtil.getConnection();
+                    PreparedStatement statement = connection.prepareStatement(query)) {
+                try (ResultSet rs = statement.executeQuery()) {
```

Comment:
do we need multiple try statements. Can't we use one


try (Connection connection = DAOUtil.getConnection();
 PreparedStatement statement = connection.prepareStatement(query);
ResultSet rs = statement.executeQuery()
) {